File: app/src/main/java/com/wugx_utils/test/TestPresenter.java
package com.wugx_utils.test;

import com.wugx_autils.mvp.presenter.BasePresenter;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;

/**
 * @author Wugx
 * @date 2018/11/8
 */
public class TestPresenter extends BasePresenter<TestContract.TestView> {

    TestContract.TestView mView;

    public TestPresenter(TestContract.TestView view) {
        mView = view;
    }

    public void showInfo(){


        Observable.timer(5, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<Long>() {
                    @Override
                    public void accept(Long aLong) throws Exception {
                        // TODO: 2018/10/8 判断登陆状态
                        String result = "我是网络请求的信息";
                        mView.showDatas(result);
                    }
                });

    }
}

File: kotlin_autils/src/main/java/com/wugx/kotlin_autils/TestPresenter.kt
package com.wugx.kotlin_autils

import com.wugx_autils.http.net.common.DefaultObserver
import com.wugx_autils.mvp.presenter.BasePresenter
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.TimeUnit

/**
 *
 *
 *@author Wugx
 *@date   2018/12/18
 */
class TestPresenter : BasePresenter<TestContract.IView>() {

    fun req() {
        rootView.showLoading("玩命加载中...")

        Observable.interval(5, TimeUnit.SECONDS)
                .subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : DefaultObserver<Long>() {
                    override fun onSuccess(response: Long?) {
                        //add disposable
                        addDisposable(mDisposable)

                        rootView.hideLoading()
                        rootView.showDatas(">>>${response}")
                    }
                })
    }

}